### PR TITLE
SONAR-7021 Merge tests metric domains

### PR DESCRIPTION
**Pull request by simonbrandhof**
SONAR-7021 Merge tests metric domains

### CODE REVIEW

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java [hunk 1]
```diff
@@ -37,7 +37,16 @@
...
+  /**
+   * @deprecated in 5.5. Merged into {@link #DOMAIN_TESTS}
+   */
+  @Deprecated
   public static String DOMAIN_INTEGRATION_TESTS = "Tests (Integration)";
+  /**
+   * @deprecated in 5.5. Merged into {@link #DOMAIN_TESTS}
+   */
+  @Deprecated
   public static String DOMAIN_OVERALL_TESTS = "Tests (Overall)";
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java [hunk 2]
```diff
@@ -37,7 +37,16 @@
   // the following fields are not final to avoid compile-time constants used by plugins
   public static String DOMAIN_SIZE = "Size";
   public static String DOMAIN_TESTS = "Tests";
+
+  /**
+   * @deprecated in 5.5. Merged into {@link #DOMAIN_TESTS}
+   */
+  @Deprecated
   public static String DOMAIN_INTEGRATION_TESTS = "Tests (Integration)";
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* INFO 2 info

Watch the comments in this conversation to review them.

**Comment by julienlancelot** [comment]
LGTM

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* INFO 2 info

Watch the comments in this conversation to review them.

**Pull request closed without merging**